src/classes/redis-connection.ts

```
@@ -48,14 +52,16 @@ export class RedisConnection extends EventEmitter {
         maxRetriesPerRequest: null,
         enableReadyCheck: false,
       };
+      this.checkUpstashHost(this.opts.host);
     } else {
       this._client = <RedisClient>opts;
-      let options = this._client.options;
+      let options = <IORedis.RedisOptions>this._client.options;
       if ((<IORedis.ClusterOptions>options)?.redisOptions) {
         options = (<IORedis.ClusterOptions>options).redisOptions;
       }
 
       this.checkOptions(deprecationMessage, options);
+      this.checkUpstashHost(options.host);
```

I think we should actually throw an exception in this case, since it is not going to work.